server/src/employee/employee.controller.ts

```diff
@@ -26,9 +19,27 @@ export class EmployeeController {
     return this.employeeService.create(createEmployeeDto);
   }
 
-
   @Get()
   getAllEmployees(): Promise<Employee[]> {
     return this.employeeService.getAllEmployees();
   }
+
+  /**
+   * TODO: Improve how error is handled
+   */
+  @Patch(':employeeId/manager')
+  async addManager(
+    @Param() { employeeId }: GetEmployeeByIdDto, 
+    @Body() { managerId }: AddManagerDto
+    ): Promise<Employee> {
+    try {
+      const manager = await this.managerService.findOne(managerId);
+
+      const employee = await this.employeeService.findOne(employeeId);
+
+      return await this.employeeService.addManager(employee, manager);
+    }catch(error: any) {
+        throw new NotFoundException
```

Comment:
```suggestion
        throw new NotFoundException()
```

il manque l'execution du construteur